enhancements/ingress/gateway-api-crd-life-cycle-management.md

```diff
@@ -0,0 +1,437 @@
+- Solve CRD life-cycle management in OLM.
+- Solve OLM subscription management.
+
+## Proposal
+
+### CRD Life-cycle management
+
+OpenShift's [Cluster Ingress Operator (CIO)] will manage the _entire_
+life-cycle of the Gateway API CRDs from here onward. The CIO will be packaged
+along with a [Validating Admission Policy (VAP)] to block updates from sources
+other than the CIO, and will hold the CRDs at a specific `$CRD_VERSION`. In
+effect, this means that the **Gateway API resources will now be treated like a
+core API.**
+
+[Cluster Ingress Operator (CIO)]:https://github.com/openshift/cluster-ingress-operator
+[Validating Admission Policy (VAP)]:https://kubernetes.io/docs/reference/access-authn-authz/validating-admission-policy/
+
+#### CRD Deployment
+
+The CIO will now check for the presence of the CRDs on the cluster, for
+which there are two scenarios:
+
+1. The CRDs are not present, so we create them
+2. The CRDs are already present, so we need to take over management
+
+The former is extremely straightforward and the correct version of the CRDs
+will be applied.
+
+The latter situation has more complexities. We'll refer to this process as "CRD
+Management Succession", and cover its implications and logic below.
+
+#### CRD Management Succession
+
+Taking over the management of the Gateway API CRDs from a previous entity
+can be tumultuous as we will require:
+
+* no other actors managing the CRDs from here on out
+* only standard CRDs can be deployed (no experimental)
+* the CRDs to be deployed at an exact version/schema
+* OR the CRDs not to be present at all
+
+We will require this by providing a pre-upgrade check in the previous release
+that verifies these are true and sets `Upgradable=false` if any of them are not.
+
+> **Note**: Upstream Gateway API unfortunately layered experimental versions of
+> CRDs on top of the same [Group/Version/Kind (GVK)] as the standard ones, so
+> unfortunately for initial release users will be unable to use experimental as
+> we can't deliver that as a part of our supported API surface. It will be
+> feasible to add a gate to enable experimental later, but because of this
+> layering this will not be a great user experience and will require the cluster
+> to become tainted. We are tracking and supporting [an effort] in upstream
+> Gateway API to separate experimental into its own group, which we expect to
+> help move towards a better overall experience for users who want experimental
+> Gateway API features going forward.
+
+**We simply can not anticipate all of the negative effects** succession will
+have on existing implementations if the cluster admin forces through the upgrade
+check. As an extra precaution for users forcing upgrades precipitously, we will
+provide an admin gate to both provide an extra warning and gather consent from
+the cluster admin to take over the management of the CRDs. This will be
+accompanied by detailed documentation on what the admin should check and how to
```

Comment:
If we use the standard "unsupported config override" for exceptions, we don't need to provide any documentation.

Reply:
```suggestion
Unsupported config override will be used for exceptions.
```